autoload/lsp/util.vim

```diff
@@ -107,8 +107,14 @@ export def LspUriRemote(uri: string): bool
   return uri =~ '^\w\+::' || uri =~ '^[a-z][a-z0-9+.-]*://'
 enddef
 
+var resolvedUris = {}
+
 # Convert a Vim filename to an LSP URI (file://<absolute_path>)
 export def LspFileToUri(fname: string): string
+  if resolvedUris->has_key(fname)
+    return resolvedUris[fname]
+  endif
+
   var uri: string = fname->fnamemodify(':p')
```

Comment:
the key of dict perhaps should be this full_path, but just a feel.

Reply:
Good point.

Reply:
Yeah, I think you're right. There is a potential issue if one would `cd` into a different directory with the same structure.